docs/docs/mapping/customizing_your_gateway.md

```diff
@@ -290,6 +290,85 @@ service Greeter {
 }
 ```
 
+### Fully Overriding Custom HTTP Responses
+
+To fully override custom HTTP responses, you can use both a Forward Response Option and a Custom Marshaller.
+
+For example with proto response message as:
+
+```proto
+message CreateUserResponse {
+  string name = 1;
+}
+```
+
+The desired HTTP response:
+
+```json5
+HTTP 200 OK
+Content-Type: application/json
+
+{"name":"John Doe"}
+```
+
+But you want to return a `201 Created` status code along with a custom response structure:
+
+```json5
+HTTP 201 Created
+Content-Type: application/json
+
+{"success":true,"data":{"name":"John Doe"}}
+```
+
+First, set up the gRPC-Gateway with the custom options:
+
+```go
+mux := runtime.NewServeMux(
+  runtime.WithMarshalerOption(runtime.MIMEWildcard, &ResponseWrapper{}),
+  runtime.WithForwardResponseOption(forwardResponse),
+)
+```
+
+Define the `forwardResponse` function to handle specific response types:
+
+```go
+func forwardResponse(ctx context.Context, w http.ResponseWriter, m protoreflect.ProtoMessage) error {
+  switch v := m.(type) {
+  case *pb.CreateUserResponse:
+    w.WriteHeader(http.StatusCreated)
+  }
+  // keep default behavior
+  return nil
+}
+```
+
+Create a custom marshaller to format the response data which utilizes the builtin JSON marshaller:
+
+```go
+type ResponseWrapper struct {
+  runtime.JSONBuiltin
+}
+
+func (c *ResponseWrapper) Marshal(data any) ([]byte, error) {
+  resp := data
+  switch v := data.(type) {
+  case *pb.CreateUserResponse:
+    // wrap the response in a custom structure
+    resp = map[string]any{
+      "success": true,
+      "data":    data,
+    }
+  }
+  // otherwise, use the default JSON marshaller
+  return json.Marshal(resp)
```

Comment:
I think you may want to use `c.JSONPb.Marshal`? It's important to use a protobuf-aware JSON marshaler with protobuf types.
```suggestion
  return c.JSONPb.Marshal(resp)
```